### PR TITLE
BLD: autoversioning and release PDF

### PR DESCRIPTION
**Pull request by lacabra**
This is so beautiful ❤️

Everytime a new commit or a new pull request is merged onto the `main` branch the GitHub Actions code in `.github/workflows/release.yml` is run which does the following:
- Checkout the code
- Create a new release automatically increasing the `patch` version following Semantic Versioning
- Generate a PDF version of the toolkit using the tag obtained in the previous step
- Upload the generated PDF to the newly created release

This code has been thoroughly tested in the [lacabra/publicgoods-toolkit](https://github.com/lacabra/publicgoods-toolkit) repo, where you can see the last couple of releases with the corresponding PDFs.

Also, thanks to the changes pushed in 3cd8884, the website will always point to the `latest` version of the PDF without having to update the link to each version every time a new release comes out.

### CODE REVIEW

**Comment by kfogel** [comment]
@lacabra Wow!  This looks great.  I think I am a reasonable choice to review it, but I am at a client event all week; can you live with it unmerged until early next week?  Or should we find a different reviewer?

(Same goes for PR #69, which obviously builds on this PR.)

**Comment by lacabra** [comment]
Yes, no rush, thanks for the heads up. Whenever you are ready.

File: .gitignore [hunk 1]
```diff
@@ -21,6 +21,7 @@ yarn-debug.log*
 yarn-error.log*
 
 # Repository-specific files
+ublicgoods-toolkit.pdf
```
Comment (kfogel) [inline .gitignore:24]: Is there a missing "p" here?

File: pdf/Makefile [hunk 2]
```diff
@@ -18,7 +18,11 @@ html: $(SOURCES)
 	@echo "HTML output is not yet implemented."
 
 # The version number is what follows the word "version" in the title.
-VERSION := "0.1.0"
+# PDF_VERSION should be passed as an environment variable, but will
+# default to 0.1.0 if not present.
+# An easy way to specify it at runtime would be to run:
+# PDF_VERSION=v0.3.0 make
+PDF_VERSION ?= "0.1.0"
```
Comment (kfogel) [inline pdf/Makefile:25]: Just curious: why the change of name from "VERSION" to "PDF_VERSION" for this variable?

Comment (lacabra) [reply pdf/Makefile:25]: Because I'm passing it as an environment variable [here](https://github.com/unicef/publicgoods-toolkit/blob/b7b8408e4edf5d6098d3e7a760414fa7fcbbabfb/.github/workflows/release.yml#L31) where I want a more descriptive name than just `VERSION` so that its purpose can be more easily identified.

File: .gitignore (hunk 1 above)
Comment (lacabra) [reply .gitignore]: Yes, good catch. Fixed it 72ce177

File: pdf/Makefile (hunk 2 above)
Comment (kfogel) [reply pdf/Makefile:25]: Got it, thanks.

**Review by kfogel — APPROVED** [review]
LGTM!
